projects/atomic-swap-service/headless-wallet.md

```diff
@@ -79,25 +85,28 @@ This will allow the wallet user to add any existing proposal to the wallet's `li
 
 To ensure the informed parameters are correct, a `get` request will also be made at this time to the service backend, and a validation executed on the received contents. Any failure found on the process will be thrown and the registration discarded.
 
-### Fetch
+### Listened Proposals endpoint
+A new route `[GET] /wallet/atomic-swap/tx-proposal/listenedProposals` will be created, allowing the user to retrieve the list of `proposalIds` currently being listened to. No password should be retrieved on this call.
+
+### Fetch endpoint
 A new route `[GET] /wallet/atomic-swap/tx-proposal/fetch/{proposalId}` will be created, allowing the user to pool the service backend for updates on a registered proposal.
 
 Calls to this route for a `proposalId` that has not been registered will raise a `404` error.
 
-### Update
+### Update endpoint
 The [current workflow](https://hathor.gitbook.io/hathor/guides/headless-wallet/atomic-swap#step-3-bob-updates-alices-partial-transaction) of updating proposals will be kept unaltered, requiring calls to the `[POST] /wallet/atomic-swap/tx-proposal` route. However, in order to persist the changes on the Atomic Swap Service, the additional body parameter `service.proposal_id` must also be informed on the request.
 
 Calling this route with a proposal identifier that has not been registered will raise a `400` error. Any errors raised while interacting with the service will also be treated and returned on the http response.
 
-### Unregister
+### Unregister endpoint
 A new route `[DELETE] /wallet/atomic-swap/tx-proposal/delete/{proposalId}` will be created, allowing the user to stop listening to this proposal for the wallet specified on the request header.
 
-Should this wallet be the last one listening to the proposal, it should also be removed from the map and its websocket connection closed.
+Also, should a wallet be stopped, all proposals it's currently listening to should also be removed, and its websocket connection closed.
 
 ### External notifications
-For every listened proposal a websocket connection will be opened with the Atomic Swap Service, informing the application about any changes that happen to it through the event `wallet:update-proposal`.
+For every wallet with a registered proposal, a websocket connection will be opened with the Atomic Swap Service, informing the application about any changes that happen to its proposals through the event `wallet:update-proposal`.
 
-Whenever a message arrive through this channel, all wallets that listen to the updated proposal will receive a notification through the _External Notifications_ feature containing the updated proposal contents.
+Whenever a message arrive through this channel, the corresponding wallet will receive a notification through the _External Notifications_ feature containing the updated proposal full contents.
```

Comment:
Do we know what are the events that will be received?
We should have at least a list of them, for documentation and to check if the names won't collide with existing events on the external notifications.

Reply:
Good point: added a list of all the events, both from the wallet lib and the ones emitted on the external notifications.